src/config.c

```diff
@@ -576,22 +576,37 @@ int git_config_foreach_match(
  * Setters
  **************/
 
-static int config_error_nofiles(const char *name)
+static int get_writable_backend(git_config_backend **out, git_config *cfg, const char *name)
 {
+	size_t i;
+	file_internal *f;
+
+	*out = NULL;
+
+	if (git_vector_length(&cfg->files) == 0) {
+		giterr_set(GITERR_CONFIG,
+			"cannot set value for '%s' when no config files exist", name);
```

Comment:
If I can bikeshed here just a touch, it might be nice to switch this message on whether we're in a `set` function or in `delete_entry`.